packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.ts

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (C) 2017 Erisson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {
+        SearchInWorkspaceServer,
+        SearchInWorkspaceOptions,
+        SearchInWorkspaceResult,
+        SearchInWorkspaceClient
+} from "../common/search-in-workspace-interface";
+import { ILogger } from "@theia/core";
+import {
+        inject,
+        injectable
+} from "inversify";
+import {
+        RawProcess,
+        RawProcessFactory,
+        RawProcessOptions
+} from '@theia/process/lib/node';
+
+import * as rg from 'vscode-ripgrep';
+
+@injectable()
+export class RipgrepSearchInWorkspaceServer implements SearchInWorkspaceServer {
+
+        /* List of ongoing searches, maps search id to a the started rg process.  */
+        private ongoingSearches: Map<number, RawProcess> = new Map();
+
+        /* Each incoming search is given a unique id, returned to the client.  This is the next id we will assigned.  */
+        private nextSearchId: number = 0;
+
+        private client: SearchInWorkspaceClient | undefined;
+
+        constructor(
+                @inject(ILogger) protected readonly logger: ILogger,
+                @inject(RawProcessFactory) protected readonly rawProcessFactory: RawProcessFactory,
+        ) { }
+
+        setClient(client: SearchInWorkspaceClient | undefined): void {
+                this.client = client;
+        }
+
+        /* Search for the string WHAT in directory ROOT.  Return the assigned search id.  */
+        search(what: string, root: string, opts?: SearchInWorkspaceOptions): Promise<number> {
```

Comment:
Will it be possible to write tests for it?

Reply:
There is `ripgrep-search-in-workspace-server.spec.ts`.  Or do you mean tests for something else?

Reply:
sorry, overlooked it, github collapsed it, too many tests :)